hydra/core/override_parser/types.py

```diff
@@ -124,6 +124,23 @@ class IntervalSweep(Sweep):
     start: Optional[float] = None
     end: Optional[float] = None
 
+    def __eq__(self, other: Any) -> Any:
+        if isinstance(other, IntervalSweep):
+            eq = (
+                self.start == other.start
+                and self.end == other.end
+                and self.tags == other.tags
+            )
+
+            st = type(self.start)
+            ost = type(other.start)
+            et = type(self.end)
+            ose = type(other.end)
+            eq = eq and st == ost and et is ose
+            return eq
```

Comment:
`eq = eq` ?

Reply:
yes

Reply:
sorry I didnt understand. What does that do?

Reply:
it computes a new boolean from the boolean eq and the expression following it, and assigns it to the variable eq.

Reply:
Ahh right - thanks!